utils/validators.js

```diff
@@ -96,22 +96,40 @@ module.exports = {
       }
     });
   },
-  validateLocationPin: (
-    locationPinId,
-    checkParanoid = false,
-    errorMessage = 'Invalid location pin ID: ' + locationPinId
-  ) => {
+  validateLocationPin: (params) => {
     const options = { paranoid: true };
-    if (checkParanoid) {
+
+    if (params.checkParanoid) {
       options.paranoid = false;
     }
-    return db.locationPins
-      .findByPk(locationPinId, options)
-      .then((locationPin) => {
-        if (!locationPin) {
-          throw new Error(errorMessage);
-        }
-      });
+
+    if (params.pinType === 'CCP') {
+      if (!params.ccpParentId) {
+        throw new Error('A CCP pin must be created by a valid CCP');
+      }
+    }
+
+    if (params.pinType === 'EVENT') {
+      if (!params.eventId) {
+        throw new Error('An event pin must be created by a valid event');
+      }
+
+      if (params.ccpParentId) {
+        throw new Error('An event pin cannot be created by a CCP');
+      }
+    }
+
+    if (!params.newPin) {
+      return db.locationPins
+        .findByPk(params.locationPinId, options)
+        .then((locationPin) => {
+          if (!locationPin) {
+            throw new Error(params.errorMessage);
+          }
+        });
+    }
+
+    return true;
```

Comment:
don't really need to return true - we only care if the validation fails and an error is thrown
```suggestion
```